### ISSUE TITLE
How can I do it in quasar?

### ISSUE THREAD

**Issue by AIRGG** [jd1378/nuxt-obfuscator#4]
Hello, I have been used your module for tailwindcss it's working like a charm. But when I try it in component like quasar or nuxt/ui it's not working. even it only basic component like button. I have this simple reproducible code [here](https://github.com/AIRGG/probable-octo-system).

as you can see the output
for "padding" and "gutter" (number 1) it's working, because I type it the class in div.
but, when I'm using "component button: q-btn" from quasar (number 2) the class it's not obfuscate, because I guess it generate by component q-btn 
![samplecode](https://github.com/jd1378/nuxt-obfuscator/assets/43308693/0141075c-d4f7-4a0f-bab2-585b4a188cba)

can you help me? 
Thanks

**Comment by jd1378** [comment]
Hi
Obfuscating happens in 2 steps, first step is done by detecting and obfuscating css classes in post css and creating a mapping, second step is to replace it in your files according to the created mapping

my guess is it's failing in the second step, but to make sure, can you check if the first step is being done correctly by searching the `obfuscation.map.json` generated file for `q-btn` ?

**Comment by AIRGG** [comment]
yeah all the class is generated, I have checked one by one, as you can see the output in [here](https://github.com/AIRGG/probable-octo-system/blob/master/obfuscation.map.json).
maybe the problem in the second step? because I think the first step is working

**Comment by jd1378** [comment]
yes, as I suspected it's the second step

sadly currently I don't know a surefire way to replace classes on the final output, however I am willing to accept contributions if you got any idea

this is the piece of code responsible for replacing classes in files:
https://github.com/jd1378/nuxt-obfuscator/blob/main/src/build/vite-plugin.ts

**Comment by AIRGG** [comment]
I have some interesting again, when I try to copy from generated inspect element, 
then I put it in code, it's working, maybe it's only working for typing class, not from generated component
![samplecode](https://github.com/jd1378/nuxt-obfuscator/assets/43308693/55b1b0d5-6155-4041-befd-e1ad0746d54c)

**Comment by jd1378** [comment]
yeah its probably a problem only for components outside the project files

**Comment by AIRGG** [comment]
is there any config we can try to make it 'post', maybe we can do something like waiting component generated first, then transform it [like said in here](https://vitejs.dev/guide/api-plugin#plugin-ordering)
is there will be problem or break something?

**Comment by jd1378** [comment]
I'm not sure, but I think I remember there was a problem with post that I didn't use it

can you try changing it locally and see how it goes for you ?

**Comment by AIRGG** [comment]
using 'post' is not working

**Comment by AIRGG** [comment]
I revert it back to 'pre', I think I know the problem, I try to check generated 'dist' folder,  yeah it's still component or function with parameter I guess (number 1), not the generated button with class. that's why your module can't transform it when build process
but in (number 2) I type it directly it's working. Yeah I think there is nothing wrong in your module

![image_2024-03-05_115745433](https://github.com/jd1378/nuxt-obfuscator/assets/43308693/df4c9197-def6-44db-8fda-a48a2e6e3617)

**Comment by jd1378** [comment]
hmm
okay
I'm sorry that I couldn't help with this
let me know if you found a way

**Comment by jd1378** [comment]
Hi @AIRGG 
can you try v2 and let me know if it's working as expected ?

**Comment by jd1378** [comment]
ok test 2.0.1
v2 uses `post`, so hopefully, it should work with everything everywhere

**Comment by AIRGG** [comment]
still not working, right now my assumption the component & the css is generated in runtime process, like I mention in here.

so when user load the page in browser, it will run function with parameter `number 1`, then the function will be generate the component like button with css, so I guess it will not work in build process, whatever pre or post

> I revert it back to 'pre', I think I know the problem, I try to check generated 'dist' folder, yeah it's still component or function with parameter I guess (number 1), not the generated button with class. that's why your module can't transform it when build process but in (number 2) I type it directly it's working. Yeah I think there is nothing wrong in your module
> 
> ![image_2024-03-05_115745433](https://private-user-images.githubusercontent.com/43308693/309987737-df4c9197-def6-44db-8fda-a48a2e6e3617.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2ODc5NjQsIm5iZiI6MTcxMDY4NzY2NCwicGF0aCI6Ii80MzMwODY5My8zMDk5ODc3MzctZGY0YzkxOTctZGVmNi00NGRiLThmZGEtYTQ4YTJlNmUzNjE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDE1MDEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllZWM4MWY2ZDFhZjFjNDFmYjhkYThhOTQ1Mzg0MjhhN2NmMjExYzc1YjU1MjY5NmU1OTBkMGU0MTVmODBkMWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.orPFYurai0k8nCF_1F2LKWOky4MhuuGK-IyO4IOsUok)
